pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java

```diff
@@ -386,14 +386,27 @@ private class SetMessageTTL extends CliCommand {
         @Parameter(description = "tenant/namespace", required = true)
         private java.util.List<String> params;
 
-        @Parameter(names = { "--messageTTL", "-ttl" }, description = "Message TTL in seconds. "
-                + "When the value is set to `0`, TTL is disabled.", required = true)
-        private int messageTTL;
+        @Parameter(names = { "--messageTTL", "-ttl" },
+                description = "Message TTL in seconds (or minutes, hours, days, weeks eg: 100m, 3h, 2d, 5w). "
+                        + "When the value is set to `0`, TTL is disabled.", required = true)
+        private String messageTTLStr;
```

Comment:
Will there be compatibility issues? Some scripts written by users cannot be used now because they do not contain units.

Reply:
No, it's compatible. If the option does not contain a unit, its value is treated as a number of seconds.